docs/tutorials/essentials/part-7-rtk-query-basics.md

```diff
@@ -0,0 +1,692 @@
+
+## Displaying Posts with Queries
+
+### Using Query Hooks in Components
+
+Now that we have the API slice defined and added to the store, we can import the generated `useGetPostsQuery` hook into our `<PostsList>` component and use it there.
+
+Currently, `<PostsList>` is specifically importing `useSelector`, `useDispatch`, and `useEffect`, reading posts data and loading state from the store, and dispatching the `fetchPosts()` thunk on mount to trigger the data fetch. **The `useGetPostsQueryHook` replaces all of that!**
+
+Let's see how `<PostsList>` looks when we use this hook:
+
+```jsx title="features/posts/PostsList.js"
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+import { Spinner } from '../../components/Spinner'
+import { PostAuthor } from './PostAuthor'
+import { TimeAgo } from './TimeAgo'
+import { ReactionButtons } from './ReactionButtons'
+
+// highlight-next-line
+import { useGetPostsQuery } from '../api/apiSlice'
+
+// highlight-next-line
+let PostExcerpt = ({ post }) => {
+  return (
+    <article className="post-excerpt" key={post.id}>
+      <h3>{post.title}</h3>
+      <div>
+        <PostAuthor userId={post.user} />
+        <TimeAgo timestamp={post.date} />
+      </div>
+      <p className="post-content">{post.content.substring(0, 100)}</p>
+
+      <ReactionButtons post={post} />
+      <Link to={`/posts/${post.id}`} className="button muted-button">
+        View Post
+      </Link>
+    </article>
+  )
+}
+
+export const PostsList = () => {
+  // highlight-start
+  const {
+    data: posts,
+    isLoading,
+    isSuccess,
+    isError,
+    error
+  } = useGetPostsQuery()
+  // highlight-end
+
+  let content
+
+  // highlight-next-line
+  if (isLoading) {
+    content = <Spinner text="Loading..." />
+    // highlight-next-line
+  } else if (isSuccess) {
+    content = posts.map(post => <PostExcerpt key={post.id} post={post} />)
+    // highlight-next-line
+  } else if (isError) {
+    content = <div>{error.toString()}</div>
+  }
+
+  return (
+    <section className="posts-list">
+      <h2>Posts</h2>
+      {content}
+    </section>
+  )
+}
+```
+
+Conceptually, `<PostsList>` is still doing all the same work it was before, but we were able to replace the multiple `useSelector` calls and the `useEffect` dispatch with a single call to `useGetPostsQuery()`.
+
+Each generated query hook returns an object containing several fields, including:
+
+- `data`: the actual response contents from the server. **This field will be `undefined` until the response is received**.
+- `isLoading`: a boolean indicating if the hook is currently making the _first_ request to the server for this data
```

Comment:
"first request from this hook instance", but not necessarily "for this data".
If you switch from page 1 to page 2 and page 2 needs to be fetched, `isLoading` will not reactivate.